tests/base.py

```diff
@@ -60,6 +60,11 @@ def setUpClass(cls):
         cls.config.read_config(config_path)
 
     def setUp(self, auth_mode='ApiUser'):
+        # When running the tests from a pull request from a client, the Shotgun
+        # site URL won't be set, so do not attempt to run the test.
+        if not self.config.server_url:
```

Comment:
This allows us to skip any tests that require a live Shotgun server when receiving a PR from a client.